package/src/completionCacheManager/completionCacheManager.ts

```diff
@@ -0,0 +1,23 @@
+import * as monaco from 'monaco-editor/esm/vs/editor/editor.api';
+import * as ls from 'vscode-languageserver-types';
+
+export const createCompletionCacheManager = (getFromLanguageService: GetFromLanguageService) => {
+    let completionList: ls.CompletionList | undefined;
+    let lastWord: string | undefined;
+
+    return {
+        getCompletionItems: async (word: string | undefined, resource: monaco.Uri, position: ls.Position) => {
+            const isIncluded = word?.includes(lastWord);
+            const shouldGetItems = !lastWord || !word || !isIncluded;
+            if (shouldGetItems) {
+                completionList = await getFromLanguageService(resource, position);
+            }
+
+            lastWord = word;
+            return Promise.resolve(completionList);
+        },
+    };
+};
```

Comment:
Does this need to be async? If we remove the `await` point we won't need to worry about how/if this affects order, of it it's been canceled.

```suggestion
export const createCompletionCacheManager = (getFromLanguageService: GetFromLanguageService) => {
    let completionList: Promise<ls.CompletionList> | undefined;
    let lastWord: string | undefined;

    return {
        getCompletionItems: (word: string | undefined, resource: monaco.Uri, position: ls.Position) => {
            const isIncluded = word?.includes(lastWord);
            const shouldGetItems = !lastWord || !word || !isIncluded;
            if (shouldGetItems) {
                completionList = getFromLanguageService(resource, position);
            }

            lastWord = word;
            return completionList;
        },
    };
};
```

Reply:
Good call. I am changing it.